src/api/dependency-discovery/src/router.js

```diff
@@ -1,15 +1,36 @@
 const { Router } = require('@senecacdot/satellite');
-const getDependencies = require('./dependency-list');
+const { getDependencyList, getNpmPackageInfo, isPackageDependency } = require('./dependency-list');
 
 const router = Router();
 
 router.get('/projects', async (req, res, next) => {
   try {
     res.set('Cache-Control', 'max-age=3600');
-    res.status(200).json(await getDependencies());
+    res.status(200).json(await getDependencyList());
   } catch (err) {
     next(err);
   }
 });
 
+// For project names that use a full name convention (e.g. @senecacdot/satellite),
+// we will use /projects/:namespace/:name, for projects that have regular name,
+// then we will use /projects/:name.
+// That means that if "name" is undefined, then we hit /projects/:name, instead.
+router.get('/projects/:namespace/:name?', async (req, res, next) => {
+  const { namespace, name } = req.params;
+  const packageName = name ? `${namespace}/${name}` : namespace;
+
+  if (!(await isPackageDependency(packageName))) {
```

Comment:
`isPackageDependency` could throw too, right?  We should wrap that in try/catch as well.